src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private Ui ui;
+    public TaskList(ArrayList<Task> tasks, Ui ui) {
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+    public TaskList(Ui ui) {
+        this.tasks = new ArrayList<>();
+        this.ui = ui;
+    }
+
+    public String filterTasks(String description) {
+        List<Task> filteredTasks = new ArrayList<>();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (task.getDescription().contains(description)) {
+                filteredTasks.add(task);
+            }
+        }
+        return this.ui.listFoundTasks(filteredTasks);
+    }
+
+    public void save(Storage storage) {
+        storage.save(this.tasks);
+    }
+
+    public String listTasks() {
+        return this.ui.listMessage(tasks);
+    }
+
+    public String markTaskAsDone(int taskNumber) {
```

Comment:
Really good naming of all your functions in your PR throughout, not just in this file.